app/demo-geant-integration/SensitiveDetector.cc

```diff
@@ -41,9 +54,29 @@ G4bool SensitiveDetector::ProcessHits(G4Step* step, G4TouchableHistory*)
         return false;
     }
 
-    auto time = step->GetPreStepPoint()->GetGlobalTime();
+    // Fill the hit data for this step
+    auto         touchable = step->GetPreStepPoint()->GetTouchable();
+    unsigned int id        = touchable->GetVolume()->GetCopyNo();
+    HitData      data{id,
+                 edep,
+                 step->GetPreStepPoint()->GetGlobalTime(),
+                 touchable->GetTranslation()};
+
+    // Add energy deposition for this cell if it was hit before
+    for (auto hit : *(collection_->GetVector()))
+    {
+        if (id == hit->data().id)
+        {
```

Comment:
Are there fewer enough IDs per SD that this linear search is OK?

Reply:
This section is removed now.  In general, the number of hit id will be roughly close to the number of physics volumes, which can be large for typical physics events.  So, your comment is still relevant if we introduce a hit reduction for the CMS-like detector.